Assets/MathsAppliqués/Vecteurs/Scripts/CircleDetection.cs

```diff
@@ -2,21 +2,25 @@
 
 public class CircleDetection : MonoBehaviour
 {
-    [SerializeField] [Range(0,10)] private float circleSize;
-    [SerializeField] CircleDetection circleDetection;
-    public float CircleSize => circleSize; 
+    [SerializeField, Range(0,10)] private float _circleSize;
+    [SerializeField] private CircleDetection _circleDetection;
+    public float CircleSize => _circleSize; 
+    
     private void OnDrawGizmos() 
     {
-        if(DistanceVector(transform.position,circleDetection.transform.position) <= (circleSize + circleDetection.CircleSize)*(circleSize + circleDetection.CircleSize) )
+        float DistanceVector(Vector2 circle, Vector2 point)
```

Comment:
Yep c'est ça !
Il est recommandé de les mettre à la fin du bloc de la fonction englobante pour justement ne pas gêner la lecture de la fonction en elle-même :
https://learn.microsoft.com/fr-fr/dotnet/csharp/programming-guide/classes-and-structs/local-functions